src/main/java/org/killbill/billing/plugin/vertex/VertexTaxCalculator.java

```diff
@@ -294,7 +294,7 @@ private SaleRequestType toTaxRequest(final Account account,
                                          @Nullable final String originalInvoiceReferenceCode,
                                          final boolean dryRun,
```

Comment:
Should we have a test for dryrun?

Reply:
From logical perspective the difference for `dryRun`/`noDryRun` for Vertex are 2 things:

1. For dryRun we use `QUOTATION` sale message type instead of `INVOICE` when constructing Vertex request which I believe affects if the tax calculation transaction will be recorded in Vertex (please see the snipper below) - not sure if it worth to test it in scope of this PR.

```
if (dryRun) {
            taxRequest.setSaleMessageType(SaleMessageTypeEnum.QUOTATION);
        } else {
            taxRequest.setSaleMessageType(SaleMessageTypeEnum.INVOICE);
        }
```
2. there is no dao.addResponse invocation for dryRun -  added test for that
```
if (!dryRun) {
                dao.addResponse(account.getId(), newInvoice.getId(), kbInvoiceItems, taxResult, clock.getUTCNow(), kbTenantId);
            }
```